### PR TITLE
feat(ui): use vite in the query-builder branch

### PR DESCRIPTION
**Pull request by gallotamas**
## Description

Support building with vite on the `misi--PA-14842--query-builder-feature` branch.
Related to the changes in this PR: https://github.com/openclarity/vmclarity/pull/1012
Adds some additional configuration to vite that is required by the changes introduced in the query builder branch.

## Type of Change

[ ] Bug Fix  
[x] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [x] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

**Review by chrisgacsal — APPROVED** [review]

File: ui/index.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    <meta charset="utf-8" />
+    <link rel="icon" href="/favicon.png" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="manifest" href="/manifest.json" />
+    <title>VMClarity</title>
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <script type="module">
```
Comment (misi-bp) [inline ui/index.html:13]: This solution--using inline script here--is going to be a new issue amongst the Solis findings. I know it's a low prior issue, but later we may have to find an other solution to remove them or make them exceptions by adding the hashes.

Comment (gallotamas) [reply ui/index.html:13]: vite transforms it automatically to an external script. You can see it on this screenshot:
image

Comment (misi-bp) [reply ui/index.html:13]: We have this discussed, Vite.js already uses hashes in prod mode, so if we need it later, we can read and use the filename in nginx to get the hash and add it to the CSP header.

**Review by misi-bp — APPROVED** [review]
Thank you, Tomi! Very nice

**Pull request merged**